gremlin-javascript/src/main/javascript/gremlin-javascript/lib/structure/graph.ts

```diff
@@ -73,20 +77,26 @@ export class Vertex<T extends Record<string, any> = Record<string, any>> extends
   }
 }
 
-export class Edge<T extends Record<string, any> = Record<string, any>> extends Element {
+export class Edge<
+  TOutVertex extends Vertex = Vertex,
+  TLabel extends string = string,
+  TInVertex extends Vertex = Vertex,
+  TProperties extends Record<string, any> = Record<string, any>,
+  TId extends number | string = number,
```

Comment:
I think this is too restrictive. Providers can use any type they like for the ID. JanusGraph for example has its own type.

Reply:
Thanks, I've removed the type restriction.